components/whyd/2022/Terminal/UserInputArea.vue

```diff
@@ -154,6 +161,13 @@ export default {
         if (typeof func === 'function') {
           func()
         }
+        if (processed.startsWith('cd')) {
+          // for cds from buttons
+          this.navigate(processed.replace('cd ', ''))
+        }
+      } else if (processed.startsWith('cd')) {
+        // for extra cds
+        this.navigate(processed.replace('cd', '').trim())
```

Comment:
the duplication here is scary and makes me worry about commands only working in button form and not in text form or vice versa. can you reorganize the logic to only check for and process this command once?

Reply:
I think I just needed to step away for a bit to figure out how to reorganize it.